src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -523,16 +522,55 @@ private function readInput($inputStream, Question $question)
         }
 
         $ret = '';
+        $cp = $this->setIOCodepage();
         while (false !== ($char = fgetc($multiLineStreamReader))) {
             if (\PHP_EOL === "{$ret}{$char}") {
                 break;
             }
             $ret .= $char;
         }
 
+        $ret = $this->resetIOCodepage($cp, $ret);
+
         return $ret;
     }
 
+    /**
+     * Set console I/O to the host code page.
+     *
+     * @return int Previous code page in IBM/EBCDIC format
+     */
+    private function setIOCodepage(): int
+    {
+        if (\function_exists('sapi_windows_cp_set')) {
+            $cp = sapi_windows_cp_get();
+            sapi_windows_cp_set(sapi_windows_cp_get('oem'));
+
+            return $cp;
+        }
+
+        return 0;
+    }
+
+    /**
+     * Set console I/O to the specified code page.
+     * Also make string conversion.
+     *
+     * @param int    $cp Code page in IBM/EBCDIC format
+     * @param string $in User input
+     *
+     * @return string
```

Comment:
You dont need these two lines.
One will understand the return type form the signature.